xilem/src/property_tuple.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use masonry_winit::core::{Properties, Property, Widget, WidgetMut};
+
+/// Helper trait implemented for all tuples of `Option<SomeProperty>` up to 12 items.
+pub trait PropertyTuple {
+    /// Helper method for [`xilem_core::View::build`].
+    ///
+    /// Returns list of properties to be set on the created widget.
+    fn build_properties(&self) -> Properties;
```

Comment:
So concretely, my suggestion is to make this be something like:
```suggestion
    fn build_properties(&self, props: &mut Properties);
```

Reply:
The `with_props` method has been removed, so now the code does `elem.properties = build_properties()`, which I think is more readable than a mutable parameter.

I'd rather merge the API as-is and maybe consider a version with a mutable param later. I don't think it changes much either way.